pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -437,6 +437,17 @@ func (s *NodeController) deletePods(nodeID string) error {
 		if err := s.kubeClient.Pods(pod.Namespace).Delete(pod.Name); err != nil {
 			glog.Errorf("Error deleting pod %v: %v", pod.Name, err)
 		}
+		if pod.Spec.RestartPolicy == api.RestartPolicyAlways {
```

Comment:
@bgrant0607 

Thanks, will close the pull request. Let me know if you would like to have the kubelet client timeout. I have seen pod cache get stuck for several minutes trying to reach a dead node.